pandas/core/missing.py

```diff
@@ -404,7 +441,13 @@ def _from_derivatives(xi, yi, x, order=None, der=0, extrapolate=False):
     return m(x)
 
 
-def _akima_interpolate(xi, yi, x, der=0, axis=0):
+def _akima_interpolate(
+    xi: np.ndarray,
+    yi: np.ndarray,
+    x: Union[Scalar, ArrayLike],
```

Comment:
i think these are just np.ndarray right?

Reply:
and similar in almost all places that you are using Union[Scalar, ArrayLike] unless you can reveal_type that this is not the case

Reply:
You're right, they're always `ndarray` except in `_interpolate_scipy_wrapper`. It's because in `_interpolate_scipy_wrapper` we cast any scalar input to `ndarray` and these functions always get called from there:
https://github.com/pandas-dev/pandas/blob/e91cbd774838ee0189b9b18cb8e234517921e0fe/pandas/core/missing.py#L306

Reply:
Fixed these, and docstrings too